v3/rfcs/dx/20240620-simplify-subgraph-getting-started-and-federation.md

```diff
@@ -0,0 +1,47 @@
+# DX: simplify subgraphs for getting started and federation workflows
+
+Status: Review required
+
+## subgraph init command
+What this is: as a part of making it easier for people to work through the getting started guide, or to scaffold project files, allow the user to go from scratch to an API with only CLI commands.
+
+init-ing a supergraph or connector will create files for both local and cloud environments. This is about what happens for the subgraphs.
+
+```bash
+# creates a subgraph folder and subgraph.yaml file
+ddn subgraph init app
+
+# add subgraph.yaml to the supergraph.yaml file and create an env file for the env
+ddn supergraph add subgraph --supergraph supergraph.local.yaml --subgraph app/subgraph.yaml
+touch app/.env.app.local
+
+# create the supergraph build
+ddn supergraph build --supergraph supergraph.yaml --env-file app/.env.app.local
+
+# deploy to cloud
+touch app/.env.app.cloud
+ddn connector build create --connector connector.yaml --target-env-file app/.env.app.cloud
```

Comment:
We might not need a touch here. If the command is like `ddn connector build create --connector mydb/connector.yaml --target-subgraph app/subgraph.yaml --target-connector-link mypg --target-env-file app/.env.app.cloud` . If the `--target-env-file` is not provided, we use the `envFile` specified in subgraph.yaml. If the flag is provided, we create the file if it does not exist and write it there

Reply:
Agree, can do without touch. Makes for better cross platform docs as well.